validator/client/health_monitor.go

```diff
@@ -0,0 +1,123 @@
+package client
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/OffchainLabs/prysm/v6/validator/client/iface"
+	"github.com/sirupsen/logrus"
+)
+
+type healthMonitor struct {
+	ctx       context.Context
+	cancel    context.CancelFunc
+	v         iface.Validator
+	maxFails  int
+	healthyCh chan bool // emits true → healthy, false → unhealthy
+	fails     int
+	isHealthy bool
+	sync.RWMutex
+}
+
+// newHealthMonitor
+func newHealthMonitor(
+	parentCtx context.Context,
+	parentCancel context.CancelFunc,
+	maxFails int,
+	v iface.Validator,
+) *healthMonitor {
+	m := &healthMonitor{
+		ctx:       parentCtx,
+		cancel:    parentCancel,
+		maxFails:  maxFails,
+		v:         v,
+		healthyCh: make(chan bool, 1),
+	}
+
+	// Prime channel with the current status so consumers have an initial value.
+	isHealthy := v.FindHealthyHost(parentCtx)
+	m.isHealthy = isHealthy
+	m.healthyCh <- isHealthy
+	if !isHealthy {
+		m.fails = 1
+	}
+
+	return m
+}
+
+func (m *healthMonitor) IsHealthy() bool {
+	m.RLock()
+	defer m.RUnlock()
+	return m.isHealthy
+}
+
+func (m *healthMonitor) performHealthCheck() {
+	m.Lock()
+	defer m.Unlock()
+
+	ishealthy := m.v.FindHealthyHost(m.ctx)
+	if ishealthy {
+		m.fails = 0
+	} else {
```

Comment:
```suggestion
	} else if m.maxFails > 0 {
```

Reply:
actually i should combine it with the bottom one too